src/rules/utils.ts

```diff
@@ -633,6 +633,37 @@ export const isHook = (
   node.callee.type === AST_NODE_TYPES.Identifier &&
   HookName.hasOwnProperty(node.callee.name);
 
+export const isFunctionDeclarationTestCase = (
+  context: TSESLint.RuleContext<string, any>,
+  node: TSESTree.FunctionDeclaration,
+): boolean => {
+  return getFunctionDeclarationTestCallExpressions(context, node).length > 0;
+};
+
+export const getFunctionDeclarationTestCallExpressions = (
+  context: TSESLint.RuleContext<string, any>,
+  node: TSESTree.FunctionDeclaration,
+): Array<JestFunctionCallExpression<TestCaseName>> => {
+  const variables = context.getDeclaredVariables(node);
```

Comment:
Since this is the only need for `context`, I think it would be better to pass the results of `getDeclaredVariables` in.